connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -172,7 +178,7 @@ private List<SourceRecord> sourceRecordsForGroup(String group) throws Interrupte
             long timestamp = System.currentTimeMillis();
             Map<TopicPartition, OffsetAndMetadata> upstreamGroupOffsets = listConsumerGroupOffsets(group);
             Map<TopicPartition, Checkpoint> newCheckpoints = checkpointsForGroup(upstreamGroupOffsets, group);
-            Map<TopicPartition, Checkpoint> oldCheckpoints = checkpointsPerConsumerGroup.computeIfAbsent(group, ignored -> new HashMap<>());
+            Map<TopicPartition, Checkpoint> oldCheckpoints = checkpointsStore.contents().computeIfAbsent(group, ignored -> new HashMap<>());
             oldCheckpoints.putAll(newCheckpoints);
```

Comment:
Move this to a new `CheckpointsStore#emitCheckpoints(Map<TopicPartition, Checkpoint>)` method